src/Mobs/Monster.cpp

```diff
@@ -286,12 +287,18 @@ void cMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 	{
 		if (m_bOnGround)
 		{
-			if (DoesPosYRequireJump((int)floor(m_Destination.y)))
+			if ((m_JumpCoolDown == 0) && DoesPosYRequireJump((int)floor(m_Destination.y)))
 			{
 				m_bOnGround = false;
-
+				m_JumpCoolDown = 20;
 				// TODO: Change to AddSpeedY once collision detection is fixed - currently, mobs will go into blocks attempting to jump without a teleport
-				AddPosY(1.2);  // Jump!!
+				AddPosY(1.6);  // Jump!!
+				SetSpeedX(3.2 * (m_Destination.x - GetPosition().x));  // Move forward in a preset speed.
+				SetSpeedZ(3.2 * (m_Destination.z - GetPosition().z));  // The numbers were picked based on trial and error and 1.6 and 3.2 are perfect.
+			}
+			else if (m_JumpCoolDown > 0)
```

Comment:
Good question, and no, we can't, with your proposal:
If m_JumpCoolDown is 0 and DoesPosYRequireJump returns 0, we'll get an m_JumpCoolDown of -1.